dockerfiles/jupyter/Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM jupyter/scipy-notebook
+
+# by default COPY creates files with root permissions, so doing all the preparation with root and chown-ing everything
+# in the end
+USER root
+COPY . /tmp/mlrun
+RUN cd /tmp/mlrun && pip install ".[api]" && cd /tmp && rm -rf mlrun
+
+WORKDIR $HOME
+
+COPY ./README.md $HOME
+COPY ./examples $HOME/examples/
+
+ARG MLRUN_CACHE_DATE=initial
+RUN git clone https://github.com/mlrun/demos.git $HOME/demos
```

Comment:
suggest to minimize layers when possible. I am not sure the limitation is relevant here but docker has 42 layers limit so I suggest when possible
`RUN git clone X && git clone Y`
instead of multiple RUNs

Reply:
I accept the minimize layers suggestion, but here I think it will hurt readability, I'll make all the env vars definition one layer